addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml

```diff
@@ -0,0 +1,97 @@
+  #@schema/nullable
+  region: ""
+  #@schema/desc "The zone used by vSphere multi-AZ feature"
+  #@schema/nullable
+  zone: ""
+  #@schema/desc "The flag that disables TLS peer verification"
+  insecureFlag: False
+  #@schema/desc "The IP family configuration"
+  #@schema/nullable
+  ipFamily: ""
+  #@schema/desc "Internal VM network name"
+  #@schema/nullable
+  vmInternalNetwork: ""
+  #@schema/desc "External VM network name"
+  #@schema/nullable
+  vmExternalNetwork: ""
+  #@schema/desc "Internal VM network cidr to be excluded."
+  #@schema/nullable
+  vmExcludeInternalNetworkSubnetCidr: ""
+  #@schema/desc "External VM network cidr to be excluded."
+  #@schema/nullable
+  vmExcludeExternalNetworkSubnetCidr: ""
+  #@schema/desc "Extra arguments for the cloud-provider-vsphere."
+  cloudProviderExtraArgs:
+    #@schema/desc "External arguments for cloud provider"
+    tls-cipher-suites: "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384"
+  nsxt:
+    #@schema/desc "A flag that enables pod routing"
+    podRoutingEnabled: false
+    routes:
+      #@schema/desc "NSX-T T0/T1 logical router path"
+      routerPath: ""
+      #@schema/desc "Cluster CIDR"
+      clusterCidr: ""
+    #@schema/desc "The username used to access NSX-T"
+    username: ""
+    #@schema/desc "The password used to access NSX-T"
+    password: ""
+    #@schema/desc "The NSX-T server"
+    #@schema/nullable
+    host: ""
+    #@schema/desc "InsecureFlag is to be set to true if NSX-T uses self-signed cert"
+    insecureFlag: false
+    #@schema/desc "RemoteAuth is to be set to true if NSX-T uses remote authentication (authentication done through the vIDM)"
+    remoteAuth: false
+    #@schema/desc "VMCAccessToken is VMC access token for token based authentification"
+    vmcAccessToken: ""
+    #@schema/desc "VMCAuthHost is VMC verification host for token based authentification"
+    vmcAuthHost: ""
+    #@schema/desc "Client certificate key for NSX-T"
+    clientCertKeyData: ""
+    #@schema/desc "Client certificate data for NSX-T"
+    clientCertData: ""
+    #@schema/desc "The certificate authority for the server certificate for locally signed certificates"
+    rootCAData: ""
+    #@schema/desc "The name of secret that stores CPI configuration"
+    secretName: "cloud-provider-vsphere-nsxt-credentials"
+    #@schema/desc "The namespace of secret that stores CPI configuration"
+    secretNamespace: "kube-system"
+  #@schema/desc "HTTP proxy setting"
+  http_proxy: ""
+  #@schema/desc "HTTPS proxy setting"
+  https_proxy: ""
+  #@schema/desc "No-proxy setting"
+  no_proxy: ""
+  #@schema/desc "Used in vsphereParavirtual mode, defines the Cluster API versions. Default: run.tanzu.vmware.com/v1alpha2."
+  clusterAPIVersion: "run.tanzu.vmware.com/v1alpha2"
```

Comment:
Seems we have `run.tanzu.vmware.com/v1alpha3` available in [t-f](https://github.com/vmware-tanzu/tanzu-framework/tree/main/apis/run/v1alpha3), please confirm the version used here.

Reply:
good catch, let's use latest the `v1alpha3`

Reply:
The version in `t-f` is for TKR, while the clusterAPIVersion is the version for Cluster API CRD. 
For uTKG we should use whatever Cluster CRD version that the current Cluster has
https://github.com/kubernetes-sigs/cluster-api/tree/main/api/v1beta1

Reply:
changed to `v1beta1`